packages/core/src/tracing/instrumentation/cloud/aws/sqs.js

```diff
@@ -0,0 +1,384 @@
+const { ENTRY, EXIT, isExitSpan } = require('../../../constants');
+const requireHook = require('../../../../util/requireHook');
+const tracingUtil = require('../../../tracingUtil');
+const queueNameRE = /\/(\w+)$/;
+
+const headerNames = {
+  traceId: 'X_INSTANA_T',
+  legacyTraceId: 'X_INSTANA_ST',
+  spanId: 'X_INSTANA_S',
+  legacySpanId: 'X_INSTANA_SS',
+  level: 'X_INSTANA_L',
+  legacyLevel: 'X_INSTANA_SL'
+};
+
+let logger = require('../../../../logger').getLogger('tracing/sqs', newLogger => {
+  logger = newLogger;
+});
+
+let isActive = false;
+
+function getQueueName(queueURL) {
+  const match = queueURL.match(queueNameRE);
+
+  if (match && match.length >= 2) {
+    return match[1];
+  }
+  return '';
+}
+
+function finishSpan(err, data, span) {
+  if (err) {
+    addErrorToSpan(err, span);
+  }
+  if (typeof data === 'string') {
+    span.data.sqs.messageId = data;
+  }
+
+  span.d = Date.now() - span.ts;
+  span.transmit();
+}
+
+function addErrorToSpan(err, span) {
+  if (err) {
+    span.ec = 1;
+    if (err.message) {
+      span.data.sqs.error = err.message;
+    } else if (typeof err === 'string') {
+      span.data.sqs.error = err;
+    }
+  }
+}
+
+// TODO: maybe we can improve this implementation
+function readAttribCaseInsensitive(attributes, key1, key2) {
+  return (
+    tracingUtil.readAttribCaseInsensitive(attributes, key1) || tracingUtil.readAttribCaseInsensitive(attributes, key2)
+  );
+}
+
+function propagateSuppression(attributes) {
+  if (!attributes || typeof attributes !== 'object') {
+    return;
+  }
+
+  attributes[headerNames.level] = {
+    DataType: 'String',
+    StringValue: '0'
+  };
+}
+
+function propagateTraceContext(attributes, span) {
+  if (!attributes || typeof attributes !== 'object') {
+    return;
+  }
+
+  attributes[headerNames.traceId] = {
+    DataType: 'String',
+    StringValue: span.t
+  };
+
+  attributes[headerNames.spanId] = {
+    DataType: 'String',
+    StringValue: span.s
+  };
+
+  attributes[headerNames.level] = {
+    DataType: 'String',
+    StringValue: '1'
+  };
+}
+
+function shimSendMessage(original) {
+  return function() {
+    if (isActive) {
+      const originalArgs = new Array(arguments.length);
+      for (let i = 0; i < originalArgs.length; i++) {
+        originalArgs[i] = arguments[i];
+      }
+
+      return instrumentedSendMessage(this, original, originalArgs);
+    }
+
+    return original.apply(this, arguments);
+  };
+}
+
+function instrumentedSendMessage(ctx, originalSendMessage, originalArgs) {
+  /**
+   * Send Message Attribues format
+   * {
+   *    ...
+   *    MessageAttributes: {
+   *      CustomAttribute: {
+   *        DataType: 'String',
+   *        StringValue: 'Custom Value'
+   *      }
+   *    }
+   * }
+   */
+  let attributes = originalArgs[0].MessageAttributes;
+
+  if (!attributes) {
+    attributes = originalArgs[0].MessageAttributes = {};
+  }
+
+  if (cls.tracingSuppressed()) {
+    propagateSuppression(attributes);
```

Comment:
Got it. This was copied from GCP, where there is no span.cancel() in the suppression bit. So maybe we should take a look there as well?
https://github.com/instana/nodejs-sensor/blob/main/packages/core/src/tracing/instrumentation/cloud/gcp/pubsub.js#L85

Reply:
Huh. You are right. Very good catch! 

For pubsub there is already a test that makes sure we do not create a span when suppression is on. And thinking about it, I now see why it works: There is a check for the parent span a few lines later (here in your code as well as in GCP pubsub). And that check will fail when suppression is on because the entry instrumentation (for example httpServer.js) will not have created a span. 

Thinking about it, we should be more explicit about this and not rely on the behaviour of the entry instrumentation. I checked other exit instrumentation (amqp, kafkaJs, httpClient) and we are not entirely consistent but most explicitly stop the processing by doing something equivalent to `return originalSendMessage.apply(ctx, originalArgs);` when suppression is detected. 

In conclusion, we should add the line here and also for pubsub. It won't change the behaviour in practice but make the code easier to understand and more explicit.